File: Leetcode80. Remove duplicates from sorted array II/Solution.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        '''
        j=1
        count=1
        n=len(nums)
        for i in range(1, n):
            if nums[i]==nums[i-1]:
                count += 1
            else:
                count = 1

            if count <= 2:
                nums[j] = nums[i]
                j += 1
        
        return j
        '''
        l,r=0,0
        while r<len(nums):
            count=1
            while r+1<len(nums) and nums[r]==nums[r+1]:
                r += 1
                count += 1

            for i in range(min(2, count)):
                nums[l]=nums[r]
                l += 1
            r += 1
        
        return l

File: Leetcode80. Remove duplicates from sorted array II/Solution.java
class Solution {
    public int removeDuplicates(int[] nums) {
        int L=0, R=0;
        int n = nums.length;

        while(R<n) {
            int count = 1;
            while((R+1)<n && (nums[R]==nums[R+1])) {
                count++;
                R++;
            }
            for(int i=0; i<Math.min(2, count); i++) {
                nums[L]=nums[R];
                L++;
            }
            R++;
        }

        return L;
    }
}

// runtime - 0 ms
class Solution {
    public int removeDuplicates(int[] nums) {
        int n=nums.length;
        int k = 2;
        for(int i=2; i<n; i++) {
            if(nums[k-2] != nums[i]) {
                nums[k] = nums[i];
                k++;
            }
        }
        return k;
    }
}
